pages/in/widget/iscn/index.vue

```diff
@@ -0,0 +1,402 @@
+      opener,
+      state,
+      url,
+    } = query;
+    let {
+      license,
+      title,
+    } = query;
+    const tags = tagsString ? tagsString.split(',') : [];
+    if (!Object.keys(query).length) {
+      return redirect('https://docs.like.co/developer/iscn/web-widget/reference');
+    }
+    if (!fingerprint) {
+      return error({ statusCode: 400, message: 'INVALID_FINGERPRINT' });
+    }
+    if (publisher) {
+      if (!ISCN_PUBLISHERS[publisher]) {
+        return error({ statusCode: 400, message: 'INVALID_PUBLISHER' });
+      }
+      ({ license } = ISCN_PUBLISHERS[publisher]);
+    }
+    if (license) {
+      if (!ISCN_LICENSES[license]) {
+        return error({ statusCode: 400, message: 'INVALID_LICENSE' });
+      }
+    } else {
+      license = ISCN_PUBLISHERS.default;
+    }
+    if (title) {
+      title = title.substring(0, 255);
+    }
+
+    return {
+      fingerprint,
+      title,
+      type,
+      tags,
+      license,
+      publisher,
+      redirectUri,
+      opener: opener && opener !== '0',
+      state,
+      url,
+    };
+  },
+  head() {
+    return {
+      title: 'ISCN',
+    };
+  },
+  computed: {
+    ...mapGetters([
+      'getUserIsRegistered',
+      'getUserId',
+      'getUserInfo',
+      'getAuthCoreNeedReAuth',
+      'getIsShowingTxPopup',
+      'getPendingTxInfo',
+    ]),
+    LIKER_LAND_URL() {
+      return LIKER_LAND_URL;
+    },
+    avatarHalo() {
+      return User.getAvatarHaloType(this.getUserInfo);
+    },
+    httpReferrer() {
+      return this.$route.query.referrer || document.referrer || undefined;
+    },
+    redirectOrigin() {
+      const url = new URL(this.redirectUri, true);
+      return url.origin;
+    },
+    windowOpener() {
+      if (!window) return null;
+      return window.opener;
+    },
+    ipfsURL() {
+      return `https://ipfs.io/ipfs/${this.fingerprint}`;
+    },
+    licenseObj() {
+      const { license } = this;
+      return {
+        displayName: this.$t(`${this.license}`),
+        url: `https://ipfs.io/ipfs/${ISCN_LICENSES[license]['/']}`,
+      };
+    },
+  },
+  async mounted() {
+    if (this.opener && !window.opener) {
+      this.$nuxt.error({ statusCode: 400, message: 'Cannot access window opener' });
+    }
+    this.isLoading = false;
+  },
+  methods: {
+    ...mapActions([
+      'popupAuthDialogInPlace',
+      'setReAuthDialogShow',
+      'doUserAuth',
+      'sendISCNSignature',
+      'setErrorMsg',
+      'closeTxDialog',
+      'prepareCosmosTxSigner',
+    ]),
+    async submitTransfer() {
+      this.isLoading = true;
+      try {
+        const { cosmosWallet } = this.getUserInfo;
+        const showDialogAction = !this.redirectUri;
+        const isWait = !!this.blocking;
+
+        const {
+          fingerprint,
+          title,
+          tags,
+          type,
+          license,
+          publisher,
+          url,
+        } = this;
+        const txHash = await this.sendISCNSignature({
```

Comment:
@williamchong here need to add a signer to pass in:
before const txHash:
const signer = await this.prepareCosmosTxSigner(); // add this line
const txHash = await this.sendISCNSignature({
          userId: this.getUserId,
          ...
          signer, // add this line
 })